jsonschema/validate.go

```diff
@@ -442,7 +442,9 @@ func (st *state) validate(instance reflect.Value, schema *Schema, callerAnns *an
 		if schema.AdditionalProperties != nil {
 			// Special case for a better error message when additional properties is
 			// false.
-			if Equal(schema.AdditionalProperties, falseSchema()) {
+			// Note: this is much faster than comparing with falseSchema using Equal.
+			isFalse := schema.AdditionalProperties.Not != nil && reflect.ValueOf(*schema.AdditionalProperties.Not).IsZero()
```

Comment:
What if `schema.AdditionalProperties.Pattern != ""`?

Reply:
Although actually, that's fine: your expression is more general than the original, because a schema with `"not": {}` is going to fail validation on everything, even if other keywords are present. Technically, though, you are not testing for "the false schema".

However, I wouldn't change the code, just add a comment.